### ISSUE TITLE
Allow using featured image as background for a group block

### ISSUE THREAD

**Issue by aristath** [WordPress/gutenberg#24660]
Came up in a [discussion in the block-themes meeting](https://wordpress.slack.com/archives/C02RP4Y3K/p1597854369209700)
Related to https://github.com/WordPress/gutenberg/issues/24634 and https://github.com/WordPress/gutenberg/issues/22081

If we want FSE themes to be able to build nice headers using the post's featured image, we need to allow the use of dynamic data as properties for blocks.
For example if we want to do something like this:

![Screen Shot 2020-08-19 at 12 30 10 PM](https://user-images.githubusercontent.com/588688/90664752-bda39a00-e253-11ea-8bdf-e7f6a4af41d0.png)

we'll need to allow the group block to have a background set to the featured image.

Ideally we'd have a method abstract enough to allow using any kind of dynamic data on many blocks, but for the purposes of FSE, background-images in group blocks are a good place to start.

**Comment by noahtallen** [comment]
Related: #14744 (though I know featured image is different from images in general), and #22724

**Comment by mtias** [comment]
> we'll need to allow the group block to have a background set to the featured image.

I don't think this is the right approach (the group block). I think it should be baked into background tools for _any block_. I also don't know if it should be abstracted to any dynamic property, at least not in UI. It could be as simple as a toggle on the background options (solid, gradient, image, video, featured image).

There's been some discussions about allowing other properties (like text) to bet set to an entity (pos title). But that also proved to be maybe not necessary if you can just insert a post title block anywhere (within a cover block, etc).

**Comment by aristath** [comment]
> I don't think this is the right approach (the group block). I think it should be baked into background tools for any block.

That's even better! :heart: 
The group block was simply the logical choice to use as a 1st step :wink:

**Comment by mtias** [comment]
Fair enough :)

**Comment by LittleBigThing** [comment]
The example screenshot from the Twenty Nineteen theme may be a special case. The theme uses the featured image as an `img` on the blog page and on a single page. This way, it can make use of responsive images by default.

Just an idea, but if an image is not strictly decorative or a background image, just superimposed with text, it could be interesting to use it as an `img` (see the `object-fit` CSS property with great support now IE support is getting abandoned) instead of a background image and profit from the responsive images and lazy-loading functionalities in WP. Maybe for another, cover-block-like block? :-)

**Comment by alaczek** [comment]
> I don't think this is the right approach (the group block). I think it should be baked into background tools for any block.

A thousand times yes! This will give us the most flexibility with the designs, without re-inventing the wheel via another block.

**Comment by iamtakashi** [comment]
> I think it should be baked into background tools for any block.

Just came here to say +1 on this approach. With the approach, visual treatments that people might want for the featured image can be done with the abilities that existing blocks already have.

**Comment by paaljoachim** [comment]
Here are some off the top of my mind suggestions.....(Not sure if I like the suggestions or not...)
But right now they seem like the most straight forward way of adding a setting to the Cover and Image Blocks.

Cover Block. I added an auto refresh icon beside the checkbox as the featured image would need to automatically replace the current background image.

![Cover-Block-set-featured](https://user-images.githubusercontent.com/5323259/94338649-48ff1080-fff4-11ea-8122-34e1805178af.jpg)


------------


Image Block. Added auto refresh icon.

![Image-Block-setting-use-featured-image](https://user-images.githubusercontent.com/5323259/94338677-79df4580-fff4-11ea-8ce4-66fb8582798c.jpg)


Here is an associated issue:
https://github.com/WordPress/gutenberg/issues/13795#issuecomment-636787215
This is about setting the first image in the page/post as featured. A setting below the featured image field.

**Comment by bfiessinger** [comment]
I would also appreciate if there would be some options to use SVG Background patterns like shown on [heropatterns.com](https://www.heropatterns.com/) or [svgbackgrounds.com](https://www.svgbackgrounds.com/)

**Comment by shaunandrews** [comment]
My first thought was to add Featured Image as an option similar to background gradients: 

<img width="275" alt="image" src="https://user-images.githubusercontent.com/191598/94480335-2875b880-01a4-11eb-80c4-06bdf0326295.png">

If we consider this could expand to include other things, like videos, perhaps we could provide `select` element for choosing:

<img width="467" alt="image" src="https://user-images.githubusercontent.com/191598/94480396-42af9680-01a4-11eb-9f8f-83e48abc3eec.png">

Similar to choosing Image, the Feature Image option would remove the focal point options and place a message on top of the preview (available depending on context):

<img width="979" alt="image" src="https://user-images.githubusercontent.com/191598/94480722-b5207680-01a4-11eb-801a-db6808cf0585.png">

**Comment by ianstewart** [comment]
> the Feature Image option would remove the focal point options

Why remove the focal point option? To provide the label?

> and place a message on top of the preview (available depending on context)

Might be easier to evaluate the image thumbnail/preview if the message was above or below? Does it need the label once set?

**Comment by shaunandrews** [comment]
> Why remove the focal point option? To provide the label?

We won't always have a featured image to display, and there wouldn't be a way to set a focal point based on a specific post ID. We could still show focal point, but it would end up being the same values across every post regardless of the actual featured image.

> Might be easier to evaluate the image thumbnail/preview if the message was above or below?

I was trying to emphasize the messaging while retaining the visual appearance of the image settings.

> Does it need the label once set?

I think the label helps explain that this data is defined elsewhere, and may not exist.

**Comment by paaljoachim** [comment]
To get a birds eye perspective. I placed the controls into the Cover Block settings. It would look something like this:
![Featured-image-in-Cover-Block](https://user-images.githubusercontent.com/5323259/94800239-9a3c4500-03e4-11eb-8073-d6bd3c89a5b4.jpg)

**Comment by BrunoAHVincent** [comment]
nearly 3 years now, any news on this? Trying to use pods for this

**Comment by paaljoachim** [comment]
Currently we still are not able to easily select the Group block. (There is an issue for it.)

While for instance the Cover block has this placeholder. (Behind the Cover block tool bar we see the Group block.)
![Screenshot 2022-08-22 at 09 27 34](https://user-images.githubusercontent.com/5323259/185863995-e5678e09-ea4e-4bba-8bd6-0cea4a75fc9b.jpg)

Perhaps it is time for adding a placeholder to the Group block?
@jasmussen @jameskoster

**Comment by jasmussen** [comment]
@paaljoachim Where's that group issue? The group block has a selectable state now in trunk.

**Comment by paaljoachim** [comment]
Hey @jasmussen Joen. 

> The group block has a selectable state now in trunk.

That is nice to hear! Will it be added to the next version of the Gutenberg plugin?

I will add in some issues and a PR which I believe are somewhat associated.

I feel when working on adding a background one should also have in the back of the mind of having a way to add a featured image. @draganescu Andrei has earlier worked on the Featured image controls for the Cover block.

(I am adding this PR by @andrewserong  in as it could in some way also touch upon the featured image.)
Add a Background Image Block Support and opt-in for the Group block
https://github.com/WordPress/gutenberg/pull/39243

Moving background tools for the Cover block from the Block Inspector to the Toolbar + extending background tools to Cover/Group blocks. By @critterverse 
https://github.com/WordPress/gutenberg/issues/28643

Group Block - add support for Background Images (Should also include the featured image)
https://github.com/WordPress/gutenberg/issues/14744

There are multiple Group block enhancements which have to do with the design tools. I am wondering if we can extend the placeholder or perhaps the Group block toolbar to add in some of the Cover block toolbar options. As it can create some consistency and connection between the Group block and the Cover block we have today.

![Screenshot 2022-08-22 at 10 17 39](https://user-images.githubusercontent.com/5323259/185873626-a904bb39-b2fa-4156-88a1-9c4947bde9f6.jpg)

**Comment by jasmussen** [comment]
The clickable state will be in v14. #40664 

The issues you shared are around background images, which is cool. Just wanted to see if there was an open ticket around the not being able to click to select an empty group block, because now you can.

**Comment by masteradhoc** [comment]
+1

**Comment by kev-mas** [comment]
+1

**Comment by supernovia** [comment]
@paaljoachim  could this option be used to "automatically use a featured image" if it exists, and otherwise use the default image as a fallback?  That would be a dream come true.

![Image](https://github.com/user-attachments/assets/7f62066c-bba6-4477-8b4c-647bca1a2f5c)

**Comment by karmatosed** [comment]
This probably needs feedback over design at this point due to conversation. I am going to triage to that.

**Comment by jasmussen** [comment]
Good discussions have happened here, so has development. Since this was opened, the generic and agnostic approach to let _any_ block use a featured image as a background is being absorbed into the Block bindings feature: #77199. @ockham would you be able to take a look at this issue and see what's possible today from that angle?
